### ISSUE TITLE
5.0.1 volfraction collision? in vesicle and multilayer_vesicle

### ISSUE THREAD

**Issue by RichardHeenan** [SasView/sasview#1442]
P(Q) for vesicle and multilayer_vesicle both have parameter volfraction, when you try to multiply by S(Q) then only radius_effective appears in the gui. 

Thus for vesicle the S(Q) has incorrect shell volfraction instead of total enclosed volume, work around is to use core_shell_sphere instead.  Perhaps we need to rename volfraction in P(Q) to something else? Or not allow an S(Q) with this model - then tell the users why in the docs.

For multilayer_vesicle the volfraction may be OK for S(Q) - needs careful check.

**Comment by RichardHeenan** [comment]
hollow_cylindr and the hollow_rectangular_prism models also have P(Q) normalised to shell volume, but will I think work fine with S(Q) as they do not have volfraction as a parameter of P(Q) , so by separately adjusting scale and volfraction any desirable parameters can be chosen for P(Q)S(Q).

**Comment by RichardHeenan** [comment]
Started to look at this, comparing core_shell_sphere, vesicle and multilayer_vesicle with and without an S(Q)
Its very confusing, and there are some major issues with the volfractions.

**Comment by butlerpd** [comment]
moving to 5.0.2 as agreed on call

**Comment by RichardHeenan** [comment]
Have investigated, and concluded that the code is actually correct ! (Though as usual it may be a little confusing to some.) So I am going to downgrade this ticket, will close it later after any discussion.

P(Q) for vesicle is normalised to the volume of the shell. When you multiply it by an S(Q), the parameter table still has volfraction shown for the shell, the volume fraction for the S(Q) is however increased internally by the appropriate volume ratio, (Rcore + Thick_shell)^3 / {(Rcore + Thick_shell)^3 - Rcore^3}, though the result of this is not shown in the gui.

Multilayer_vesicle is normalised to the total outer volume, in the same way as core_shell_sphere, so volfraction is the same for P(Q_) and S(Q). 

For both of these, volfraction is shown in the gui only within the P(Q) section of the gui table. It might be better to have an volfraction_effective appear in the S(Q) section. There could be some horrible issues here for complex sum & product models.

We ought to check the documentation for these models.

To check all this I generated some test data using core_shell_sphere with and without a hardsphere S(Q), then fit them using vesicle and multilayer vesicle, the latter with three layers having very small solvent gaps to give the same overall shell thickness. I did not attempt to check the effects of polydispersity.
